src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,147 @@
+package duke.command;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.Scanner;
+
+
+public class Duke {
+    public static void listOut(Task[] tasks, int taskNumber) {
+        for (int i = 0; i < taskNumber; i++) {
+            System.out.println((i + 1) + "." + tasks[i]);
+        }
+    }
+
+    public static void markDone(String line, Task[] tasks) {
+        try {
+            int index = Integer.parseInt(line.substring(5)) - 1;
+            tasks[index].setDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println("  " + tasks[index].getStatusIcon() + " " + tasks[index].getDescription());
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a mark-done cannot be empty.");
+        }
+    }
+
+    public static boolean addTodo(String line, Task[] tasks, int taskNumber) {
+        try {
+            tasks[taskNumber] = new ToDo(line.substring(5));
+            taskNumber++;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + tasks[taskNumber - 1]);
+            System.out.println("Now you have " + taskNumber + " tasks in the list");
+            return true;
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+            return false;
+        }
+    }
+
+    public static boolean addDeadline(String line, Task[] tasks, int taskNumber) {
+        String[] words = line.split(" ");
+        int index = 0;
+        String deadlineDescription = "";
+        String by = "";
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals("/by")) {
+                index = i;
+                break;
+            }
+        }
+        if (index == 0) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+            return false;
+        }
+        for (int i = 1; i < index; i++) {
+            deadlineDescription = deadlineDescription + words[i] + " ";
+        }
+        for (int i = index + 1; i < words.length; i++) {
+            by = by + words[i] + " ";
+        }
+        tasks[taskNumber] = new Deadline(deadlineDescription, by);
+        taskNumber++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks[taskNumber - 1]);
+        System.out.println("Now you have " + taskNumber + " tasks in the list");
+        return true;
```

Comment:
You may want to consider splitting this method into multiple smaller methods that do smaller tasks and call it from this method. This goes the same for the other tasks-related methods.